go/util/crypto.go

```diff
@@ -101,7 +102,7 @@ func GenerateSaltsAndKey(token types.Token, secret string) types.SaltsAndKey {
 	hashable := sha3.Sum512([]byte(tokenStr))
 	seed := append([]byte(tokenStr), hashable[:]...)
 	seedInt := binary.BigEndian.Uint64(seed)
-	random := rand.New(rand.NewSource(int64(seedInt)))
+	random := mathrand.New(mathrand.NewSource(int64(seedInt)))
 
 	return types.SaltsAndKey{
 		Sp: GetRandomStringOfLength(keySize, random),
```

Comment:
there's some interesting things going on here, maybe it's intended, but you're re-using the random reader on each `GetRandomStringOfLength` call below, so the ordering of these objects in the struct matters...maybe this is intended, but wanted to check

Reply:
Yep, that's intended. It's how we deterministically derive these values.